### ISSUE TITLE
BUG: groupby specification with a multi-index

### ISSUE THREAD

**Issue by jreback** [blaze/dask#772]
```
In [25]: import dask, pandas

In [26]: dask.__version__
Out[26]: '0.7.1'

In [27]: pandas.__version__
Out[27]: u'0.17.0rc2'

In [28]: df = DataFrame(np.random.randn(27,2),columns=list('AB'),index=pd.MultiIndex.from_product([range(3),list('abc'),date_range('20130101',periods=3)],names=['first','second','third']))

In [29]: df
Out[29]: 
Out[29]: 
                                A         B
first second third                         
0     a      2013-01-01 -0.995226  0.429083
             2013-01-02 -0.389231  0.490488
             2013-01-03 -1.566451 -3.684522
      b      2013-01-01  0.859339  1.467608
             2013-01-02 -0.097775 -1.294470
             2013-01-03 -2.217993  1.509466
      c      2013-01-01 -0.260645  1.072987
             2013-01-02  2.098468 -1.833434
             2013-01-03 -2.009588  0.100071
1     a      2013-01-01 -0.465310  0.123307
             2013-01-02 -0.858893  0.163779
             2013-01-03  0.878080 -0.666197
      b      2013-01-01  2.152374 -2.989408
             2013-01-02 -0.010038 -1.673479
             2013-01-03  0.039144 -0.856958
      c      2013-01-01  0.380194  0.622898
             2013-01-02 -0.176932 -0.834284
             2013-01-03 -0.743904 -0.680626
2     a      2013-01-01 -1.238129 -0.846745
             2013-01-02 -0.574737  1.923274
             2013-01-03 -0.930619 -0.420236
      b      2013-01-01  0.097219  0.433632
             2013-01-02  1.399933 -0.308837
             2013-01-03  0.314898  0.129944
      c      2013-01-01  1.226088  1.175610
             2013-01-02  0.592359 -0.478968
             2013-01-03  1.125887  0.443795

In [30]: import dask.dataframe as dd

In [31]: ddf = dd.from_pandas(df.reset_index(),npartitions=3)

In [32]: ddf.groupby('second').sum().compute()
Out[32]: 
In [33]: ddf = dd.from_pandas(df,npartitions=3)

In [34]: ddf
Out[34]: dd.DataFrame<from_pandas-74162949fa8c8c7b7334ea1103670bbf, divisions=((0, 'a', Timestamp('2013-01-01 00:00:00', offset='D')), (1, 'a', Timestamp('2013-01-01 00:00:00', offset='D')), (2, 'a', Timestamp('2013-01-01 00:00:00', offset='D')), (2, 'c', Timestamp('2013-01-03 00:00:00', offset='D')))>
```

This seems buggy
```
In [35]: ddf.groupby(level='second').sum().compute()
TypeError: groupby() takes exactly 2 arguments (1 given)
```

Going back to the reset frame
```
In [36]: ddf = dd.from_pandas(df.reset_index(),npartitions=3)

In [37]: ddf.groupby('second')
Out[37]: <dask.dataframe.core.GroupBy at 0x107637890>

In [38]: ddf.groupby('second').apply(lambda x: x+1)
Out[38]: dd.DataFrame<map-partitions-2ea801f0ed8496e0c5ba4aaf5697ff98, divisions=(0, 9, 18, 26)>

In [39]: ddf.groupby('second').apply(lambda x: x+1).compute()
TypeError: 
Remote Exception:
-----------------
Could not operate 1 with block values cannot concatenate 'str' and 'int' objects
```
Seems like a nice exception (excluding nuisance columns) except that this operation *does* work for built-ins to dask, e.g. ``.sum()`` above.

**Comment by cowlicks** [comment]
@jreback the `In [35]` line seems right. There is no `'second'` level because the dataframe was created with a reset index. Are you suggesting the exception to be more consistent with pandas?

**Comment by cowlicks** [comment]
I'm not sure what your last comment means. When you replace the `lamba x: x+1` with a builtin like `sum`? This does not work for me.

**Comment by jreback** [comment]
@cowlicks no, see ``[33]`` which uses the original df (with levels) to create a new ``ddf``, it DOES have a level. I think that the ``level`` keyword is not supported on ``ddf.groupby``?

**Comment by jreback** [comment]
in ``[39]`` I think you need to ignore columns which give an exception (e.g. strings with a numeric operation)

**Comment by cowlicks** [comment]
@jreback Ah you created a dask.dataframe with a multiindex. As far as I know we do NOT support this (right @mrocklin ?). We should be clear about this in the documentation. I'm not sure what it would take to add this.

Also the code I pushed in #805 depends on there not being `MultiIndex`s.

**Comment by jreback** [comment]
@cowlicks yep ,totally cool w/o supporting a multi-index. But should raise a nice message (*or* maybe just reset), though then you can odd errors.

**Comment by mrocklin** [comment]
Correct that we don't currently support multi-indexes

**Comment by mrocklin** [comment]
I think that raising an error at the right time would be sufficient to close this issue.

**Comment by cowlicks** [comment]
Closed by #816
